pkg/context/helpers.go

```diff
@@ -51,8 +51,8 @@ func NewTestContext(t *testing.T) *TestContext {
 
 func NewTestCommand() CommandBuilder {
 	return func(command string, args ...string) *exec.Cmd {
-		testArgs := append([]string{command}, args...)
-		cmd := exec.Command(os.Args[0], testArgs...)
+		// testArgs := append([]string{command}, args...)
+		cmd := exec.Command(command, args...)
```

Comment:
I don't understand purpose of `NewTestCommand`. I see, it changes command path to `$(pwd)/debug.test`. But for plugin execution to work, cmd path should be correct

Reply:
Hmmm, I'm not sure on the genesis of this code either...

Reply:
NewTestCommand redirects commands executed by porter back to the test binary so that it can handle the command and assert that it 

1. was called
2. the proper arguments were passed

It works in conjunction with the TestMain functions declared in some of our test packages.

Otherwise executing a command from porter during a unit test, like calling a mixin, would try to call real binaries on the file system and you would be testing the binary's behavior. This wasn't what we wanted for unit tests.

We disable this behavior in some places in porter, for example in the integration tests:

https://github.com/deislabs/porter/blob/b22ec2bdbacaceb8a24521a47a145380cd94e4d1/pkg/porter/helpers.go#L68-L73

Reply:
Got it. I don't see any mocked mixin execution to mock plugin execution. So, I decided to add it as integration test